GameServerLib/GameObjects/AttackableUnits/AI/Minion.cs

```diff
@@ -103,6 +103,10 @@ public override void Update(float diff)
                         KeepFocusingTarget(); // attack/follow target
                     }
                 }
+                else if (this is ILaneMinion lm)
```

Comment:
It doesn't look clean to me. Minion should have no knowledge about LaneMinion (I think this implicated from Liskov substitution principle). In this case you want a function like `AIMove` that will do the walk to destination in LaneMinion but will do other things in Minion class.

I think that you can put:
```
if (ScanForTargets()) // returns true if we have a target
                {
                    if (!RecalculateAttackPosition())
                    {
                        KeepFocusingTarget(); // attack/follow target
                    }
                }
```
in a function (call it `AIMove` or something like that in Minion class, and then override it in LaneMinion to be:
```
base.AIMove()
lm.WalkToDestination(); // walk to destination (or target)
```